packages/trader/src/Modules/Trading/Containers/trade.jsx

```diff
@@ -26,139 +32,107 @@ const BottomWidgetsMobile = ({ tick, digits, setTick, setDigits }) => {
     return null;
 };
 
-class Trade extends React.Component {
-    state = {
-        digits: [],
-        tick: {},
-        try_synthetic_indices: false,
-    };
-
-    componentDidMount() {
-        this.props.onMount();
-    }
+const Trade = props => {
+    const [digits, setDigits] = React.useState([]);
+    const [tick, setTick] = React.useState({});
+    const [try_synthetic_indices, setTrySyntheticIndices] = React.useState(false);
+    const [is_digits_widget_active, setIsDigitsWidgetActive] = React.useState(false);
 
-    componentWillUnmount() {
-        this.props.onUnmount();
-    }
+    React.useEffect(() => {
+        props.onMount();
+        return () => props.onUnmount();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
-    componentDidUpdate(prevProps) {
-        if (isMobile() && prevProps.symbol !== this.props.symbol) {
-            this.setState({ digits: [] });
-        }
-        if (prevProps.symbol !== this.props.symbol) {
-            this.setState({ try_synthetic_indices: false });
+    React.useEffect(() => {
+        if (isMobile()) {
+            setDigits([]);
         }
-    }
+        setTrySyntheticIndices(false);
+    }, [props.symbol, setDigits, setTrySyntheticIndices]);
 
-    bottomWidgets = ({ digits, tick }) => {
-        return (
-            <BottomWidgetsMobile
-                digits={digits}
-                tick={tick}
-                setTick={t => this.setState({ tick: t })}
-                setDigits={d => this.setState({ digits: d })}
-            />
-        );
+    const bottomWidgets = ({ digits: d, tick: t }) => {
+        return <BottomWidgetsMobile digits={d} tick={t} setTick={setTick} setDigits={setDigits} />;
     };
 
-    onChangeSwipeableIndex = index => {
+    const onChangeSwipeableIndex = index => {
         if (index === 0) {
-            this.props.setMobileDigitView(true);
+            props.setMobileDigitView(true);
         } else {
-            this.props.setMobileDigitView(false);
+            props.setMobileDigitView(false);
         }
-        this.setState({
-            is_digits_widget_active: index === 0,
-        });
+        setIsDigitsWidgetActive(index === 0);
     };
 
-    onTrySyntheticIndicesClick = () => {
-        this.setState(
-            {
-                try_synthetic_indices: true,
-            },
-            () => {
-                this.setState({
-                    try_synthetic_indices: false,
-                });
-            }
-        );
+    const onTrySyntheticIndicesClick = () => {
+        setTrySyntheticIndices(true);
+        setTimeout(() => setTrySyntheticIndices(false));
```

Comment:
@shayan-binary-2 Good time to fix this issue?

Reply:
@salar-fs I believe so. @balakrishna-binary as you see we pass a boolean name `try_synthetic_indices` to trigger `MarketSelctor` dropdown open/close but I believe as we perform that by checking if `active_category` params are filled or not

Reply:
@balakrishna-binary please ignore my above comment, this PR will have some conflict with card https://trello.com/c/3Zg2VpNl/2913-salarshayan-market-close-notification-on-contract-details, and think it would be better to let @salar-fs handle this change

Reply:
Hey guys! I mean the need to use `timeOut` and reverting the value immediately. As this is a refactor I thought it might be a good place to handle this. I remember once @reza-fs raised this and @shayan-binary-2 said we need to fix this behavior.